src/rules/extensions.js

```diff
@@ -225,13 +231,32 @@ module.exports = {
             node: source,
             message:
               `Missing file extension ${extension ? `"${extension}" ` : ''}for "${importPathWithQueryString}"`,
+            ...props.fix && extension ? {
+              fix(fixer) {
+                return fixer.replaceText(
+                  source,
+                  JSON.stringify(`${importPathWithQueryString}.${extension}`),
+                );
+              },
+            } : {},
           });
         }
       } else if (extension) {
         if (isUseOfExtensionForbidden(extension) && isResolvableWithoutExtension(importPath)) {
           context.report({
             node: source,
             message: `Unexpected use of file extension "${extension}" for "${importPathWithQueryString}"`,
+            ...props.fix
+              ? {
+                fix(fixer) {
+                  return fixer.replaceText(
+                    source,
+                    JSON.stringify(
+                      importPath.slice(0, -(extension.length + 1)),
```

Comment:
See https://github.com/un-ts/eslint-plugin-import-x/pull/327/files#r2094298890

`extension` is not always at the end.

Reply:
https://github.com/un-ts/eslint-plugin-import-x/blob/813a4a087aa614a5577fe01c7f135317992d31ca/src/rules/extensions.ts#L410C11-L410C77

I just copied your implementation.

Reply:
https://github.com/un-ts/eslint-plugin-import-x/blob/813a4a087aa614a5577fe01c7f135317992d31ca/src/rules/extensions.ts#L410C11-L410C77

I believe you are also still using `importPath.slice(0, -(extension.length + 1)),`
I have implemented your last recommendations.

Thanks